projects/exui-common-lib/src/lib/components/generic-filter/generic-filter.component.ts

```diff
@@ -55,96 +58,237 @@ export class GenericFilterComponent {
+        this.form.get(field.findPersonField).get('name').setValue(null);
+        this.form.get(field.findPersonField).get('knownAs').setValue(null);
+      }
+    }
+  }
+
+  public cancelFilter(): void {
+    this.buildForm(this.config, this.settings, true);
+    if (this.config && this.config.cancelSetting) {
+      this._settings.fields = JSON.parse(JSON.stringify(this.config.cancelSetting.fields));
+    }
+    this.filterService.persist(this.settings, this.config.persistence);
+  }
+
+  public updatePersonControls(values: any, field: FilterFieldConfig): void {
+    const keys = Object.keys(values);
+    for (const key of keys) {
+      if (this.form.get(field.name) && this.form.get(field.name).get(key)) {
+        this.form.get(field.name).get(key).patchValue(values[key]);
+      }
     }
-    this._settings = {
-      id: this.config.id,
-      fields: this.selected
-    };
   }
 
   private mergeDefaultFields(filter: FilterSetting): void {
-    const fields = this.filterService.get(this.config.id) ? this.filterService.get(this.config.id).fields : filter.fields;
-    this.selected = fields;
-    filter.fields = fields;
+    if (filter) {
+      filter.fields = this.filterService.get(this.config.id) ? this.filterService.get(this.config.id).fields : filter.fields;
+    } else if (this.config && this.config.cancelSetting) {
+      this._settings = {
+        id: this.config.id,
+        fields: JSON.parse(JSON.stringify(this.config.cancelSetting.fields))
+      };
+    }
   }
 
   private getSettings(): void {
     this._settings = this.filterService.get(this.config.id);
   }
 
-  private checkFieldsConstraints(selected: { name: string, value: string[] }[]): boolean {
-    let hasErrors = false;
-
-    for (const fieldConstraint of this.config.fields) {
-      if (fieldConstraint.minSelected > 0 || fieldConstraint.maxSelected > 0) {
-        const field = selected.find((f) => f.name === fieldConstraint.name);
-        if (!field) {
-          // display error for minimum selected
-          fieldConstraint.displayMinSelectedError = true;
-          hasErrors = true;
-          this.filterService.givenErrors.next(fieldConstraint.minSelectedError);
-          continue;
+  private buildForm(config: FilterConfig, settings: FilterSetting, reset?: boolean): void {
+    const findPersonControl = this.form ? this.form.get('findPersonControl') : null;
+    this.form = this.fb.group({});
+    if (findPersonControl) {
+      // in order to maintain find person component, control needs to be kept
+      this.form.addControl('findPersonControl', findPersonControl);
+    }
+    for (const field of config.fields) {
+      if (field.type === 'checkbox') {
+        const formArray = this.buildCheckBoxFormArray(field, settings);
+        this.form.addControl(field.name, formArray);
+      } else {
+        const validators: ValidatorFn[] = [];
+        if (field.minSelected && field.minSelected > 0) {
+          validators.push(Validators.required);
```

Comment:
Shouldn't this also use the minSelected validator? (especially if field.minSelected > 1)

Reply:
Well minSelected is for checkboxes, where you might be required to have a minimum or maximum selection. Other input types don't have min selection. However if you want to make the field required adding a minSelected will ensure the field is required. I guess I should have added a `validators` field to the config. That way you can attach any number of validators.